aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -1061,31 +1068,11 @@ Tensor _test_serialization_subcmul(const Tensor& self, const Tensor& other, cons
   return self - (other * alpha);
 }
 
-Tensor& heaviside_out(const Tensor& self, const Tensor& values, Tensor& result) {
-  TORCH_CHECK(!self.is_complex() && !result.is_complex() && !values.is_complex(),
-              "heaviside is not yet implemented for complex tensors.");
-  TORCH_CHECK(self.dtype() == values.dtype() &&  result.dtype() == self.dtype(),
-              "heaviside is not yet implemented for tensors with different dtypes.");
-
-  auto iter = TensorIterator::binary_op(result, self, values);
-  heaviside_stub(iter.device_type(), iter);
-  return result;
-}
-
-Tensor heaviside(const Tensor& self, const Tensor& values) {
-  TORCH_CHECK(!self.is_complex() && !values.is_complex(),
-              "heaviside is not yet implemented for complex tensors.");
-  TORCH_CHECK(self.dtype() == values.dtype(),
-              "heaviside is not yet implemented for tensors with different dtypes.");
-
-  Tensor result;
-  auto iter = TensorIterator::binary_op(result, self, values);
-  heaviside_stub(iter.device_type(), iter);
-  return iter.output();
-}
-
-Tensor& heaviside_(Tensor& self, const Tensor& values) {
-  return at::heaviside_out(self, self, values);
+TORCH_IMPL_FUNC(heaviside_out) (
+  const Tensor& self, const Tensor& other, const Tensor& result
+) {
+  TORCH_CHECK(!result.is_complex(), "heaviside is not yet implemented for complex tensors.");
```

Comment:
you don't need this, the meta function will have run the check

Reply:
This check has been merged into the meta function